user.tf

```diff
@@ -11,3 +11,18 @@ resource "google_sql_user" "admin_user" {
 
   deletion_policy = null
 }
+
+resource "random_password" "legacy_users" {
+  for_each = toset(var.legacy_users)
+  length   = 48
+}
+
+resource "google_sql_user" "legacy_users" {
+  for_each = toset(var.legacy_users)
+
+  instance        = google_sql_database_instance.instance.name
+  name            = each.value
+  password        = random_password.legacy_users[each.value].result
+  type            = ""
```

Comment:
Based on [docs](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_user#type) this should be `"BUILT_IN"`?

Reply:
see line 10